view/next-project/src/components/common/Input/Input.tsx

```diff
@@ -11,23 +11,37 @@ interface Props {
   onChange?: (e: React.ChangeEvent<HTMLInputElement>) => void;
   children?: React.ReactNode;
   type?: string;
+  datalist?: {
+    key: string;
+    data: { id: number; name: string }[];
+  };
 }
 
 function Input(props: Props): JSX.Element {
   const className =
     'rounded-full border border-primary-1 py-2 px-4' +
     (props.className ? ` ${props.className}` : '');
   return (
-    <input
-      className={clsx(s.input, className)}
-      placeholder={props.placeholder}
-      id={props.id}
-      value={props.value}
-      onChange={props.onChange}
-      type={props.type}
-    >
-      {props.children}
-    </input>
+    <div>
+      <input
+        className={clsx(s.input, className)}
+        placeholder={props.placeholder}
+        id={props.id}
+        value={props.value}
+        onChange={props.onChange}
+        type={props.type}
+        list={props.datalist?.key}
+      >
+        {props.children}
+      </input>
+      {props.datalist?.key && props.datalist?.data && (
+        <datalist id={props.datalist.key}>
+          {props.datalist.data.map((option) => (
+            <option key={option.id} value={option.name} />
+          ))}
+        </datalist>
+      )}
```

Comment:
datalistが存在すればkeyもdataも存在すると思うので、datalistの存在だけ確認すればいいと思います
```suggestion
      {props.datalist && (
        <datalist id={props.datalist.key}>
          {props.datalist.data.map((option) => (
            <option key={option.id} value={option.name} />
          ))}
        </datalist>
      )}
```